js/client-battle.js

```diff
@@ -1117,20 +1117,61 @@
 		toggleIgnoreSpects: function (e) {
 			this.battle.ignoreSpects = !!e.currentTarget.checked;
 			this.battle.add('Spectators ' + (this.battle.ignoreSpects ? '' : 'no longer ') + 'ignored.');
+			var $messages = $('.battle-log').find('[class*="chatmessage-"]').not('.chatmessage-' + this.battle.p1.id + ', .chatmessage-' + this.battle.p2.id);
```

Comment:
This breaks if a player changes name.

Better to filter by whether the rank is star.